### PR TITLE
Ensure that the FileSettingService closes the config file input stream

### PR DESCRIPTION
**Pull request by ChrisHegarty**
Minor change to restructure the try-with-resources so that the underlying file input stream is guaranteed to have its close method called.

I've not diagnosed this as the root cause of intermittent failures we're seeing on Windows, but rather eliminating it as a potential suspect. Additionally, the countdown latch should always be decremented, even if the error handler throws, as this has been noticed on a few stack traces from blocked processes, e.g.

```
16:50:19   2> "TEST-FileSettingsServiceIT.testSettingsApplied-seed#[4A2D28AE60BF1D89]" ID=12791 WAITING on java.util.concurrent.CountDownLatch$Sync@31b54ba6
16:50:19   2> 	at java.base@18.0.2/jdk.internal.misc.Unsafe.park(Native Method)
16:50:19   2> 	- waiting on java.util.concurrent.CountDownLatch$Sync@31b54ba6
16:50:19   2> 	at java.base@18.0.2/java.util.concurrent.locks.LockSupport.park(LockSupport.java:211)
16:50:19   2> 	at java.base@18.0.2/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:715)
16:50:19   2> 	at java.base@18.0.2/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1047)
16:50:19   2> 	at java.base@18.0.2/java.util.concurrent.CountDownLatch.await(CountDownLatch.java:230)
16:50:19   2> 	at app//org.elasticsearch.reservedstate.service.FileSettingsService.stopWatcher(FileSettingsService.java:276)
16:50:19   2> 	- locked org.elasticsearch.reservedstate.service.FileSettingsService@7ff93397
16:50:19   2> 	at app//org.elasticsearch.reservedstate.service.FileSettingsService.doStop(FileSettingsService.java:120)
16:50:19   2> 	at app//org.elasticsearch.common.component.AbstractLifecycleComponent.stop(AbstractLifecycleComponent.java:63)
16:50:19   2> 	- locked org.elasticsearch.common.component.Lifecycle@24d0ddfb
```

relates #88806

### CODE REVIEW

**Review by rjernst — APPROVED** [review]
LGTM. Note that this should go back to the 8.4 branch, and so I think it should be marked as `>non-issue` and the changelog removed, since this isn't a released bug.

**Review by grcevski — APPROVED** [review]
LGTM!

File: server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java [hunk 1]
```diff
@@ -303,15 +302,20 @@ private WatchKey enableSettingsWatcher(WatchKey previousKey, Path settingsDir) t
     CountDownLatch processFileSettings(Path path, Consumer<Exception> errorHandler) throws IOException {
         CountDownLatch waitForCompletion = new CountDownLatch(1);
         logger.info("processing path [{}] for [{}]", path, NAMESPACE);
-        try (var json = new BufferedInputStream(Files.newInputStream(path))) {
-            try (XContentParser parser = JSON.xContent().createParser(XContentParserConfiguration.EMPTY, json)) {
-                stateService.process(NAMESPACE, parser, (e) -> {
+        try (
```
Comment (mark-vieira) [inline server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java:305]: This is certainly more explicit but I don't think this is actually anymore _correct_. Closing the `BufferedInputStream` should close the delegate `FileInputStream` implicitly but as you say, it's worth ruling things out.

**Review by mark-vieira — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java (hunk 1 above)
Comment (ChrisHegarty) [reply server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java:305]: Agreed.

**Comment by ChrisHegarty** [comment]
@elasticmachine run elasticsearch-ci/part-1-windows

**Pull request merged**